tests/pytests/unit/modules/test_win_useradd.py

```diff
@@ -0,0 +1,344 @@
+    _username = random_string("test-account-", uppercase=False)
+    try:
+        yield _username
+    finally:
+        try:
+            user.delete(_username, purge=True, force=True)
+        except Exception:  # pylint: disable=broad-except
+            # The point here is just system cleanup. It can fail if no account was created
+            pass
+
+
+@pytest.fixture
+def username_int():
+    _username = random_string("", uppercase=False, lowercase=False, digits=True)
+    try:
+        yield _username
+    finally:
+        try:
+            user.delete(_username, purge=True, force=True)
+        except Exception:  # pylint: disable=broad-except
+            # The point here is just system cleanup. It can fail if no account was created
+            pass
+
+
+@pytest.fixture
+def account_str(username_str):
+    with pytest.helpers.create_account(username=username_str) as account:
+        user.addgroup(account.username, "Users")
+        yield account
+
+
+@pytest.fixture
+def account_int(username_int):
+    with pytest.helpers.create_account(username=username_int) as account:
+        user.addgroup(account.username, "Users")
+        yield account
+
+
+def test_add_str(username_str):
+    ret = user.add(name=username_str)
+    assert ret is True
+    assert username_str in user.list_users()
+
+
+def test_add_int(username_int):
+    ret = user.add(name=username_int)
+    assert ret is True
+    assert username_int in user.list_users()
+
+
+def test_addgroup_str(account_str):
+    ret = user.addgroup(account_str.username, "Backup Operators")
+    assert ret is True
+    ret = user.info(account_str.username)
+    assert "Backup Operators" in ret["groups"]
+
+
+def test_addgroup_int(account_int):
+    ret = user.addgroup(account_int.username, "Backup Operators")
+    assert ret is True
+    ret = user.info(account_int.username)
+    assert "Backup Operators" in ret["groups"]
+
+
+def test_chfullname_str(account_str):
+    ret = user.chfullname(account_str.username, "New Full Name")
+    assert ret is True
+    ret = user.info(account_str.username)
+    assert ret["fullname"] == "New Full Name"
+
+
+def test_chfullname_int(account_int):
+    ret = user.chfullname(account_int.username, "New Full Name")
+    assert ret is True
+    ret = user.info(account_int.username)
+    assert ret["fullname"] == "New Full Name"
+
+
+def test_chgroups_single_str(account_str):
+    groups = ["Backup Operators"]
+    ret = user.chgroups(account_str.username, groups=groups)
+    assert ret is True
+    ret = user.info(account_str.username)
+    groups.append("Users")
+    assert salt.utils.data.compare_lists(ret["groups"], groups) == {}
+
+
+def test_chgroups_single_int(account_int):
+    groups = ["Backup Operators"]
+    ret = user.chgroups(account_int.username, groups=groups)
+    assert ret is True
+    ret = user.info(account_int.username)
+    groups.append("Users")
+    assert salt.utils.data.compare_lists(ret["groups"], groups) == {}
```

Comment:
Why are we using this compare list salt utility?

Couldn't we compare against the sorted lists?

Reply:
The comparison didn't work when I was using assert list1 == list2... I'll try it again

Reply:
```
AssertionError: assert ['Users', 'Backup Operators', 'Guests'] == ['Backup Operators', 'Guests', 'Users']
```

Reply:
Used `groups.sort()` and `ret["gruops"].sort()` and removed the `salt.utils.data` import